src/cluster_resources.rs

```diff
@@ -58,6 +61,10 @@ pub trait ClusterResource:
     fn maybe_mutate(self, _strategy: &ClusterResourceApplyStrategy) -> Self {
         self
     }
+
+    fn pod_spec(self) -> Option<PodSpec> {
```

Comment:
Maybe this helps avoiding unnecessary clones introduced in 597e751e37540dd7d4c3192b7d786043048f0f9c
```suggestion
    fn pod_spec(&self) -> Option<PodSpec> {
```

Reply:
3fc8f1d takes `&self` and clones inside the method.

Reply:
Continued below with suggestion